history.cpp

```diff
@@ -0,0 +1,68 @@
+/**
+ * \file
+ * <!--
+ * Copyright 2015 Develer S.r.l. (http://www.develer.com/)
+ * -->
+ *
+ * \brief History class implementation
+ *
+ * \author Aurelien Rainone <aurelien@develer.org>
+ */
+
+#include "history.h"
+
+History::History(QObject *parent)
+    : QObject(parent),
+      max_entries(5)
+{
+    current = -1;
+}
+
+void History::add(const QString &entry)
+{
+    // remove eventual duplicates of new entry
+    int dup = history.indexOf(entry);
+    if (dup >= 0)
+        history.removeAt(dup);
```

Comment:
Ti faccio solo notare che parli di `duplicateS` quando in realtà, per costruzione, ce ne sarà al più uno soltanto: l'invariante non scritta di questa classe infatti è che i duplicati non entrano perché sono filtrati proprio da questo codice, che è anche l'unico modo di aggiungere entry.